### DISCUSSION TITLE
Format for Map Groups with OpenID Connect

### DISCUSSION THREAD

**Discussion by mcmdev** [requarks/wiki#5782] [Help / Questions]
I'm trying to set up Wiki.js with Authentik and would like to have groups from Authentik synced to Wiki.js, but whenever I enable
map groups and try to login, I get this error message: Cannot read properties of undefined (reading 'map')

Authentik does provide a groups claim, but I believe that it's not compatible with what Wiki.js needs, and I couldn't find any documentation on what it is supposed to look like.

This is the format that Authentik provides:
`
{
    "groups": [
        "authentik Admins",
        "Administrators",
        "Staff Managers"
    ]
}
`

What should it look like?

**Comment by shawnweeks** [comment]
I'm facing the same issue and it doesn't actually seem to be the group parsing itself that's a problem. I'm adding some log statements to `server/modules/authentication/oidc/authentication.js` and it's failing after it captures the groups from the claim so I think Authentik isn't at fault here and there is a bug in the auth module. I'm going to keep poking at it and see if I can find the issue.

**Comment by shawnweeks** [comment]
@MCMDEV I was able to get it working and there does seem to be a bug in the OIDC authentication. To fix this you need to change two lines of code in `server/modules/authentication/oidc/authentication.js` and restart the app.

First line 35 should look like this `const currentGroups = (await user.$relatedQuery('groups').select('groups.id')).map(g => g.id)` All I did was remove `.groups` immediately before `.map`

Second there seems to be a bug on line 41 but you'll only encounter it if you remove a users group. The syntax for `.unrelate()` seems incorrect and `await user.$relatedQuery('groups').unrelate().where("groupId",groupId)` fixes the issue in my testing.

**Comment by shawnweeks** [comment]
Someone already beat me to it. See https://github.com/requarks/wiki/pull/5781

**Comment by bwesselb** [comment]
@MCMDEV , 
I noticed with 2.5.291 that White Space Characters are a bad idea.   
I fixed by groups names by making them urlsafe. Replace the Whitespace with `%20`

**Comment by MarcProe** [comment]
Hi,

I have a groups claim format like "group1|group2|group3", and it does not seem to apply the groups. So the question still might be, what format is needed?

Thanks!

Edit: I tried ["group1","group2","group3"], still no luck. Seems the string is not recognized as JSON-Array.
Edit 2: adding `const groups = JSON.parse(_.get(profile, '_json.' + conf.groupsClaim))` to [authentication.js](https://github.com/requarks/wiki/blob/main/server/modules/authentication/oidc/authentication.js#L36) solves the problem. Does'nt feel right to do that, though.
